### PR TITLE
chore: rename the `lookup` crate to `path`

### PR DESCRIPTION
**Pull request by fuchsnj**
Now that lookup V1 is gone, this re-organizes the crate a bit (removing the `lookup_v2` prefix) and renames the entire crate to `path` since the term "lookup" isn't used anymore.

### CODE REVIEW

File: lib/path/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#![deny(warnings)]
```
Comment (spencergilbert) [inline lib/path/src/lib.rs:1]: Are we working on moving this up to apply to everything in the project?

**Review by spencergilbert — APPROVED** [review]
Looks like a straight forward PR, left one question and wanted to make sure we don't need to make changes to Vector - this is all internal to VRL, not the public interface?

**Comment by fuchsnj** [comment]
> wanted to make sure we don't need to make changes to Vector
This will definitely require changes in Vector. I have another PR coming out soon that will do some more re-organization of the crates, then I was planning to pull in those changes back into Vector.

File: lib/path/src/lib.rs (hunk 1 above)
Comment (fuchsnj) [reply lib/path/src/lib.rs:1]: I'm not really convinced either way yet on this. I think it makes sense on binaries where the dependencies are locked, but for a library, you can have dependency versions change at any time that could start to generate warnings and fail CI. I'm just keeping this as-is for now until it's more clear how to proceed here.

Comment (spencergilbert) [reply lib/path/src/lib.rs:1]: Fair enough, I was definitely thinking of things from how we've been doing Vector

**Pull request merged**